### ISSUE TITLE
Depth Anything: Unleashing the Power of Large-Scale Unlabeled Data

### ISSUE THREAD

**Issue by kabupen** [kabupen/papers#3]
# Abstract

データ収集を自動化することで大規模データの収集に成功した。

# Introduction

単眼深度推定（MDE）は大規模データセットの作成が困難であることから発展が遅れていた。MiDaS はデータ収集の方法などを工夫することで精度をあげた先行研究である。ただしデータの偏りなどから zero-shot での推論は精度が制限されていた。本論文ではデータセットを拡張することで、MDE の基盤モデルの作成を目指した。

データの収集方法は

- センサーの利用（LiDAR）
- ステレオマッチング
- SfM

などがあったが、コスト面では劣っていた。ここではラベルなし画像を収集しているのだが、以下のような利点がある

- simple and cheap to accquire：簡単かつ安価な取得: モノクロ画像は至る所に存在し、特別な装置を必要とせずに簡単に集めることができる
- diverse：より幅広いシーンをカバーすることができ、汎化性能が期待できる
- easy to annotate：pre-trained MDE モデルを用いてラベルなし画像をアノテーションできるため効率的である

## MiDaS について

MiDaS モデルの再現を行い、学習方法はそちらを参考にしていた。MiDaS の問題点はそもそも深度データセットが少なかったため複数のデータセットを用いる必要があったことから、　深度のスケールの違いなどに対応する必要があったことである。データセットはMiDaS 論文 Table.1 で示されているように

![image](https://github.com/kabupen/papers/assets/19812756/26a7c913-1fe5-4a88-8c54-df0946c0b353)

様々な形式の ground truth であることに対処しなければならなかった。絶対距離、unkown scale、disparity maps（視差マップ）などなど。

# DepthAnything

## Learning Labeled Images

affine invariant loss の使用。予測深度と真値との差分を全ピクセルで計算する。

$$
L = \frac{1}{N_{pix}} \sum_{i}^{N_{pix}} \rho(d_i^\star, d_i)
$$

## Unleashing the Power of Unlabeled Images

とりあえずデータを収集して、既存モデルで擬似ラベルを振っていく。教師モデルを $T$ としたときに

$$
D^u = \{ (u_i, T(u_i)) \}_{i=1}^N
$$

のようにデータラベルを自動で作成し、これで生徒モデル $S$ を学習した。どうやら $T$ でスタートするよりも $S$ から始めたほうがよりよい性能になったらしい。

### パイロット版の失敗

少量のラベル付きデータと疑似ラベルデータを混ぜて学習して、$T,S$ の学習サイクルを行うような手法は初め上手く行かなかったとのこと。（状況にはよるが）本論文では

> We conjecture that, with already sufficient labeled images in our case, the extra knowledge acquired from additional unlabeled images is rather limited.

今回のタスクでは十分なラベル付きデータが存在していて、その上で新規の疑似ラベルデータで学習しようとしたときに知見獲得に制限がかかっていた。$S$ と $T$ で同じモデル構造をしていたので、正解・不正解の傾向が同じになってしまい学習が進まなかったらしい。

cf.
- 73 [S++](https://arxiv.org/abs/2106.05095)
- 55 [FixMatch](https://arxiv.org/ftp/arxiv/papers/2001/2001.07685.pdf)

![image](https://github.com/kabupen/papers/assets/19812756/bd6d2014-7f0e-4f72-ae7c-3e49150053ce)

### 今回はどうした？

そこで、student model の学習のために、さらに難しいタスクを用意することにした。学習中に強い augmentation を入れた

- CutMix
- Gaussian blurring

こうすることでより積極的に学習が進むとした。

## Semantic-Assisted Perception

セマンティックセグメンテーションの情報を用いることで深度推定がより精度が向上するという先行研究に倣い、DINOv2 を用いた。

$$
L = 1 - \frac{1}{N} \sum \cos (f_i, f_i\prime)
$$

Depth 推定した特徴量と DINOv2の特徴量のコサイン類似度を測定する。

## パイプライン概要

![image](https://github.com/kabupen/papers/assets/19812756/347051da-c7b4-4fc0-951d-fcb7bac210fa)

- labeled/unlabeld image のパイプラインがそれぞれ実線、点線で表現されている
- sup は supervised かな
# Conclusion

# 感想

「This work was done during an internship at TikTok」はバケモノだ。。。
